delfin/drivers/hpe/hpe_3par/component_handler.py

```diff
@@ -211,3 +214,198 @@ def list_volumes(self, context):
+                    iscsi_map = ports_iscsi_map.get(port_id)
+                    if iscsi_map:
+                        ip_addr = iscsi_map.get('ipaddr')
+                        ip_mask = iscsi_map.get('netmask/prefixlen')
+                        rate = iscsi_map.get('rate')
+                if not ip_addr and ports_rcip_map:
+                    rcip_map = ports_rcip_map.get(port_id)
+                    if rcip_map:
+                        ip_addr = rcip_map.get('ipaddr')
+                        ip_mask = rcip_map.get('netmask')
+                        rate = rcip_map.get('rate')
+                if not ip_addr and port_fs_map:
+                    fs_map = port_fs_map.get(port_id)
+                    if fs_map:
+                        ip_addr = fs_map.get('ipaddr')
+                        ip_mask = fs_map.get('netmask')
+                        rate = fs_map.get('rate')
+                if not rate and ports_fcoe_map:
+                    fcoe_map = ports_fcoe_map.get(port_id)
+                    if fcoe_map:
+                        rate = fcoe_map.get('rate')
+                if ip_addr and ip_addr != '-':
+                    pattern = re.compile(consts.IPV4_PATTERN)
+                    search_obj = pattern.search(ip_addr)
+                    if search_obj:
+                        ipv4 = ip_addr
+                        ipv4_mask = ip_mask
+                    else:
+                        ipv6 = ip_addr
+                        ipv6_mask = ip_mask
+                wwn = None
+                mac = None
+                if port_type.upper() == 'ETH':
+                    mac = port.get('port_wwn/hw_addr')
+                else:
+                    wwn = port.get('port_wwn/hw_addr')
+                port_model = {
+                    'name': port_id,
+                    'storage_id': storage_id,
+                    'native_port_id': port_id,
+                    'location': port_id,
+                    'connection_status':
+                        consts.PORT_CONNECTION_STATUS_MAP.get(
+                            port.get('state', '').upper(),
+                            constants.PortConnectionStatus.UNKNOWN),
+                    'health_status': constants.PortHealthStatus.NORMAL,
+                    'type': consts.PORT_TYPE_MAP.get(port_type.upper(),
+                                                     constants.PortType.OTHER),
+                    'logical_type': None,
+                    'speed': self.analyse_speed(rate),
+                    'max_speed': self.analyse_speed(max_speed),
+                    'native_parent_id': None,
+                    'wwn': wwn,
+                    'mac_address': mac,
+                    'ipv4': ipv4,
+                    'ipv4_mask': ipv4_mask,
+                    'ipv6': ipv6,
+                    'ipv6_mask': ipv6_mask,
+                }
+                port_list.append(port_model)
+        return port_list
+
+    def analyse_speed(self, speed_value):
```

Comment:
Can we change name to parse_speed ? analyse is confusinf name

Reply:
Modified